datafusion/proto/src/logical_plan/from_proto.rs

```diff
@@ -586,7 +586,10 @@ pub fn parse_expr(
                 parse_exprs(&pb.args, registry, codec)?,
                 pb.distinct,
                 parse_optional_expr(pb.filter.as_deref(), registry, codec)?.map(Box::new),
-                parse_vec_expr(&pb.order_by, registry, codec)?,
+                match pb.order_by.len() {
+                    0 => None,
+                    _ => Some(parse_exprs(&pb.order_by, registry, codec)?),
+                },
```

Comment:
this pattern is already used here
https://github.com/apache/datafusion/blob/117ab1b8c861f431b2d04443e2028ca16fafc57c/datafusion/proto/src/logical_plan/mod.rs#L817-L819

Reply:
I wonder if in both places, we could use another new helper like

```rust
fn optimize_empty_vec<T>(v: Vec<T>) -> Option<Vec<T>> {
    (!v.empty()).then_some(v)
}
```

so the combination would look like this:

```rust
optimize_empty_vec(parse_exprs(&pb.order_by, registry, codec)?)
```

However, that's a personal preference, mostly based on the fact that naming the input (`pb.order_by`) twice, which can easily introduce copy&paste errors.

Reply:
`empty_vec_to_none`? `vec_if_non_empty`?

in this case, however, my preference would be to remove `Option` and keep just `Vec`.
Empty collection perfectly describes lack of sorting, so no need to wrap if with optional value.

Reply:
>  Empty collection perfectly describes lack of sorting, so no need to wrap if with optional value.

Let's try to do that then. I agree that having an `Option<Vec<_>>` in `new_udf` doesn't make much sense.

Reply:
good idea. i would rather have this as a follow-up though.
this work would conflict with https://github.com/apache/datafusion/pull/12177 considerably.

Reply:
filed https://github.com/apache/datafusion/issues/12195 issue for this